ffi/wgpu.h

```diff
@@ -126,8 +128,25 @@ typedef enum {
 } WGPUPowerPreference;
 
 typedef enum {
+  /**
+   * The presentation engine does **not** wait for a vertical blanking period and
+   * the request is presented immediately. This is a low-latency presentation mode,
+   * but visible tearing may be observed. Will fallback to `Fifo` if unavailable on the
+   * selected  platform and backend. Not optimal for mobile.
+   */
```

Comment:
I think this should be VSync vs NoVSync

Reply:
We used to have that, but then #512 came and it seems reasonable to have mailbox. It doesn't affect portability, and it's very useful.

Reply:
Ok, we could add them to webgpu.h. It's basically:

 - vsync off
 - vsync on with double buffering
 - vsync on with triple buffering

?

Reply:
You are basically suggesting to treat Vsync as Mailbox all the time? That seems appealing at first. I wonder if there are reasons for users to not use Mailbox when it's available.

Reply:
Actually I meant that we could change webgpu.h to have the three modes and was trying to understand what the semantic of them are. But saying VSync is Mailbox and FIFO when not available would work too.

Reply:
I wouldn't describe `Mailbox` in those terms. With traditional vsync you are capped at your monitors refresh rate (e.g. 60 FPS). This throttles your event loop and can create input lag. 

With Mailbox there is no cap or throttling. It's more like `Immediate` mode than it is `Fifo` in that aspect. Frames can be acquired and submitted as fast as possible. If multiple frames are submitted before the next refresh, the older frames may be discarded. It's low latency like `Immediate` mode but prevents tearing like `Fifo`. This does however, comes at a cost which is increased power consumption.

For reference:

https://developer.samsung.com/sdp/blog/en-us/2019/07/26/vulkan-mobile-best-practice-how-to-configure-your-vulkan-swapchain

> The app can acquire a new image straight away, render to it, and present it. If an image is queued for presentation, it will be discarded. Mobile demands efficiency; hence, the word "discarded" should be a big red flag when developing on mobile – the aim should always be to avoid unnecessary work.
> 
> Since an image was queued for presentation, the framerate will not improve. **What is the advantage of MAILBOX then? Being able to keep submitting frames lets you ensure you have the latest user input, so input latency can be lower versus FIFO.**
> 
> The price you pay for MAILBOX can be very steep. If you don't throttle your CPU and GPU at all, one of them may be fully utilized, resulting in higher power consumption. Unless you need low-input latency, our recommendation is to use FIFO.